src/main/java/io/vertx/core/json/JsonStructure.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright (c) 2011-2017 The original author or authors
+ * ------------------------------------------------------
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * and Apache License v2.0 which accompanies this distribution.
+ *
+ *     The Eclipse Public License is available at
+ *     http://www.eclipse.org/legal/epl-v10.html
+ *
+ *     The Apache License v2.0 is available at
+ *     http://www.opensource.org/licenses/apache2.0.php
+ *
+ * You may elect to redistribute this code under either of these licenses.
+ */
+package io.vertx.core.json;
+
+import io.vertx.core.shareddata.impl.ClusterSerializable;
+
+/**
+ * A representation of a <a href="http://json.org/">JSON</a> structure in Java.
+ * 
+ * JSON is built on two structures:
+ *
+ * A collection of name/value pairs. In various languages, this is realized as
+ * an object, record, struct, dictionary, hash table, keyed list, or associative
+ * array. An ordered list of values. In most languages, this is realized as an
+ * array, vector, list, or sequence.
+ *
+ * @author Jan Zajic
+ *
+ */
+public abstract class JsonStructure implements ClusterSerializable {
```

Comment:
I can think of a case where I would like to extend it, for example on JDBC client, the result set is a JSON object however the column names returned are case insensitive, however since we encode to json we must get a column by the name the RDBMS returns. A extension could be a JSON that uses a case insensitive key for the underlying map for this case.

But anyway this would be a pure hack for this specific case and would only work as long as it wasn't encoded over the EB or copied.